src/viewport.js

```diff
@@ -159,11 +171,11 @@ export default class Viewport {
     const z0 = coord0[2];
     const z1 = coord1[2];
 
-    const t = z0 === z1 ? 0 : (targetZ - z0) / (z1 - z0);
+    const t = z0 === z1 ? 0 : ((targetZ || 0) - z0) / (z1 - z0);
     const v = vec2_lerp([], coord0, coord1, t);
 
     const vUnprojected = this.unprojectFlat(v);
-    return xyz.length === 2 ? vUnprojected : [vUnprojected[0], vUnprojected[1], 0];
+    return Number.isFinite(targetZ) ? [vUnprojected[0], vUnprojected[1], targetZ] : vUnprojected;
```

Comment:
cause targetZ's default is not set to 0, here we won't return [lat, lon, 0] while user doesn't set targetZ, which is not consistent with what we say in doc.

Reply:
The document can explain this better, yes.
Essentially `unproject([x, y])` is our most common use case (in hover and click callbacks), and we are keeping it consistent with the old API which returns `[lng, lat]`.

Reply:
Understood we need  keep it the same as old API, so in doc line" `opts.targetZ` | `Number`  | `0`      | If pixel depth is missing, use as the desired elevation |", maybe we could say default value is 'undefined', cause it is not '0' anymore.
'